walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -62,18 +62,46 @@
     }
     
     List<Event> mandatoryAttendeeEvents = new ArrayList<Event>();
+    List<Event> optionalAttendeeEvents = new ArrayList<Event>();
     for (Event event : allEvents) {
       if (!optionalAttendeesOnly(event, mandatoryRequestAttendees)) {
         mandatoryAttendeeEvents.add(event);
+      } else {
+        optionalAttendeeEvents.add(event);
       }
     }
     sortByStart(mandatoryAttendeeEvents);
+    sortByStart(optionalAttendeeEvents);
     
-    if (!mandatoryAttendeeEvents.isEmpty()) {
-      return getAllPossibleTimes(mandatoryAttendeeEvents, request);
+    if (mandatoryAttendeeEvents.isEmpty()) {
```

Comment:
This doesn't seem right, and you should probably add an explicit test for this. The test will need to have no events with mandatory attendees and then a set of possible event times with that give a sub-optimal number of optional attendees.

Reply:
This was based on the tests we were asked to write previously for adding optional attendees. If there are only optional attendees, they will all be considered equally.

Reply:
Misunderstood, added a test for this case that now fails.